editor/src/clj/internal/util.clj

```diff
@@ -660,3 +660,15 @@
                                    all-reduced reduced)))
                        (mapv transient tos)
                        from)))))))
+
+(defn make-dedupe-fn
+  "Given a sequence of non-unique items, returns a function that will return the
+  same instance of each item if it is determined to be equal to an item already
+  in the ever-growing set of unique items."
+  [items]
+  (let [unique-volatile (volatile! (set items))]
+    (fn dedupe-fn [item]
+      (or (get @unique-volatile item)
+          (do
+            (vswap! unique-volatile conj item)
+            item)))))
```

Comment:
We use this to split the string we get back from the Lua preprocessors into lines in an attempt to keep duplicate strings out of the cache since a lot of the lines in a preprocessed file will be kept the same.

Reply:
I don't understand how it works... Is it basically the same as `identity`? Just returns the input... Or is it for ensuring that for equal items we preserve only identical ones?

If it's for ensuring identical values, I'm not sure `get` from the set works:
```clj
(let [string-set #{(str "fo" "o")}
      s "foo"]
  {:equal (= (first string-set) s)
   :identical (identical? (first string-set) s)
   :get (identical? s (get string-set s))})
=> {:equal true
    :identical false
    :get false}
```

Reply:
It's intended to make sure you get a reference to the original item when it is equal to the value you put in. I think your `:get` check is wrong? It should be:
```clj
(let [string-set #{(str "fo" "o")}
      s "foo"]
  {:equal (= (first string-set) s)
   :identical (identical? (first string-set) s)
   :get (identical? (first string-set) (get string-set s))})
=> {:equal true,
    :identical false,
    :get true}
```
I wrote a unit test for it in `internal/util_test.clj` that you can see further down in the pull request.

Reply:
Yes, my mistake, that makes sense!